pkg/platform/kube/test/platform_test.go

```diff
@@ -52,6 +58,87 @@ type DeployFunctionTestSuite struct {
 	KubeTestSuite
 }
 
+func (suite *DeployFunctionTestSuite) TestDeployFromGit() {
+	branch := "refs/heads/test-deploy-from-git"
+
+	functionIngressHost := testk8s.GetDefaultIngressHost()
+
+	createFunctionOptions := suite.CompileCreateFunctionOptions("reverser-from-git")
+	createFunctionOptions.FunctionConfig.Spec.ServiceType = v1.ServiceTypeClusterIP
+	createFunctionOptions.FunctionConfig.Spec.Build.FunctionSourceCode = "" // empty source code so it'll build from git
+	createFunctionOptions.FunctionConfig.Spec.Build.Path = "http://localhost:5050/org/repo.git"
+	createFunctionOptions.FunctionConfig.Spec.Build.CodeEntryType = build.GitEntryType
+	createFunctionOptions.FunctionConfig.Spec.Build.CodeEntryAttributes = map[string]interface{}{
+		"branch":  branch,
+		"workDir": "common/reverser/python",
+		"gitCredentials": map[string]interface{}{
+			"accessToken": "goodAccessToken",
+		},
+	}
+	createFunctionOptions.FunctionConfig.Spec.Handler = "reverser:handler"
+	createFunctionOptions.FunctionConfig.Spec.Runtime = "python"
+	createFunctionOptions.FunctionConfig.Spec.Triggers = map[string]functionconfig.Trigger{
+
+		// create http trigger with ingress, so we can invoke the function using it later (for testing on minikube)
+		"httptrig": {
+			Kind: "http",
+			Attributes: map[string]interface{}{
+				"ingresses": map[string]interface{}{
+					"0": map[string]interface{}{
+						"host":  functionIngressHost,
+						"paths": []string{"/"},
+					},
+				},
+			},
+		},
+	}
+
+	service := gitkit.New(gitkit.Config{
```

Comment:
move it to a file, add `+build <labels>` and import it from test, use from the test that uses it
in other words, make it a client for test suites.